libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp

```diff
@@ -598,4 +598,64 @@ void AC_AttitudeControl_Heli::set_inverted_flight(bool inverted)
         _transition_count = AC_ATTITUDE_HELI_INVERTED_TRANSITION_TIME * AP::scheduler().get_filtered_loop_rate_hz();
     }
     _inverted_flight = inverted;
+
+}
+
+// Command a thrust vector and heading rate
+void AC_AttitudeControl_Heli::input_thrust_vector_rate_heading(const Vector3f& thrust_vector, float heading_rate_cds, bool slew_yaw)
+{
+    // convert thrust vector to a roll and pitch angles
+    // this negates the advantage of using thrust vector control, but works just fine
+    Vector3f angle_target = attitude_from_thrust_vector(thrust_vector, _ahrs.yaw).to_vector312();
+
+    float euler_roll_angle_cd = degrees(angle_target.x) * 100.0f;
+    if (_inverted_flight) {
+        euler_roll_angle_cd = wrap_180_cd(euler_roll_angle_cd + 18000);
+        AC_AttitudeControl::input_euler_angle_roll_pitch_euler_rate_yaw(euler_roll_angle_cd, degrees(angle_target.y) * 100.0f, heading_rate_cds);
+    } else {
+        AC_AttitudeControl::input_thrust_vector_rate_heading(thrust_vector, heading_rate_cds, slew_yaw);
+    }
+
 }
+
+// Command a thrust vector, heading and heading rate
+void AC_AttitudeControl_Heli::input_thrust_vector_heading(const Vector3f& thrust_vector, float heading_angle_cd, float heading_rate_cds)
+{
```

Comment:
I think is better to get the easy bit out of the way first.
```suggestion
{
    if (!_inverted_flight) {
            AC_AttitudeControl::input_thrust_vector_heading(thrust_vector, heading_angle_cd, heading_rate_cds);
            return;
    }
```
The pattern of doing the pass through case and returning means its very easy to see that this only changes behavior in the inverted flight case.

Reply:
done.